Libraries/LMCommon/LanguageModel.swift

```diff
@@ -0,0 +1,160 @@
+// Copyright © 2024 Apple Inc.
+
+import Foundation
+@preconcurrency import Hub
+import MLX
+import MLXNN
+import Tokenizers
+
+extension Module {
+
+    /// Compute the number of parameters in a possibly quantized model
+    public func numParameters() -> Int {
+        return leafModules().flattenedValues().map {
+            mod -> Int in
+            if let qlin = mod as? QuantizedLinear {
+                return qlin.scales.size * qlin.groupSize
+            } else if let qemb = mod as? QuantizedEmbedding {
+                return qemb.scales.size * qemb.groupSize
+            } else {
+                return mod.parameters().flattenedValues().reduce(
+                    0,
+                    {
+                        $0 + $1.size
+                    })
+            }
+        }.reduce(0, +)
+    }
+}
+
+public struct LMInput {
```

Comment:
A new union type that holds the different inputs to `generate()` and `LanguageModel.prepare()`